### PR TITLE
feat(swap): prevent caching of old swap amount after success swap

### PR DESCRIPTION
**Pull request by milan-bc**
## Description (optional)
This PR prevents caching of swap initial value, after success swap we now do clear previously entered amount

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/swap/sagas.ts [hunk 1]
```diff
@@ -222,6 +222,7 @@ export default ({
       }
       yield put(actions.form.stopSubmit('previewSwap'))
       yield put(actions.components.refresh.refreshClicked())
+      yield put(actions.form.destroy('swapAmount'))
```
Comment (dsisto-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/components/swap/sagas.ts:225]: May or may not help: I think this is automatically done when unmounting the reduxForm component. Maybe your component has `destroyOnUnmount: false`.

**Review by dsisto-bc — APPROVED** [review]
LGTM! But probably someone with more knowledge on the impact of wallet forms should have a look

File: packages/blockchain-wallet-v4-frontend/src/data/components/swap/sagas.ts (hunk 1 above)
Comment (schnogz) [reply packages/blockchain-wallet-v4-frontend/src/data/components/swap/sagas.ts:225]: i believe it does, @milan-bc can you check that. removing that would be the correct way to fix this

**Review by schnogz — CHANGES_REQUESTED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/components/swap/sagas.ts (hunk 1 above)
Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/swap/sagas.ts:225]: we do have it `destroyOnUnmount: false` but what I notice then if you go forward and want to return back to enter amount page we lost entered amount in that case, that is why I added destroy after we do create order

Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/swap/sagas.ts:225]: we have preview swap where user can decide to go back, UX thingy we have to keep form value on unmount in such case

**Pull request merged**